File: sfas_mobile/lib/features/vendors/models/vendor.dart
class Vendor {
  final String id;
  final String name;
  final String contactEmail;
  final String phoneNumber;
  final String taxId;
  final String address;

  Vendor({
    required this.id,
    required this.name,
    required this.contactEmail,
    required this.phoneNumber,
    required this.taxId,
    required this.address,
  });

  factory Vendor.fromJson(Map<String, dynamic> json) {
    return Vendor(
      id: json['id'] ?? '',
      name: json['name'] ?? '',
      contactEmail: json['contactEmail'] ?? '',
      phoneNumber: json['phoneNumber'] ?? '',
      taxId: json['taxId'] ?? '',
      address: json['address'] ?? '',
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'name': name,
      'contactEmail': contactEmail,
      'phoneNumber': phoneNumber,
      'taxId': taxId,
      'address': address,
    };
  }
}

File: sfas-backend/src/main/java/com/sfas/sfas_backend/domain/entity/Vendor.java
package com.sfas.sfas_backend.domain.entity;

import jakarta.persistence.*;
import lombok.*;
import org.hibernate.annotations.UuidGenerator;
import org.jspecify.annotations.NonNull;
import org.jspecify.annotations.Nullable;
import org.springframework.data.annotation.CreatedDate;
import org.springframework.data.annotation.LastModifiedDate;
import org.springframework.data.jpa.domain.support.AuditingEntityListener;

import java.time.LocalDateTime;
import java.util.UUID;

@Entity
@Table(name = "vendors")
@EntityListeners(AuditingEntityListener.class)
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Vendor {

    @Id
    @GeneratedValue
    @UuidGenerator
    @Nullable
    private UUID id;

    @Column(nullable = false)
    @NonNull
    private String name;

    @Column(nullable = false)
    @NonNull
    private String contactEmail;

    @Column(nullable = false)
    @NonNull
    private String phoneNumber;

    @Nullable
    private String taxId;

    @Nullable
    private String address;

    @ManyToOne(fetch = FetchType.LAZY, optional = false)
    @JoinColumn(name = "created_by_id", nullable = false)
    private User createdBy;

    @CreatedDate
    @Column(updatable = false)
    @Nullable
    private LocalDateTime createdAt;

    @LastModifiedDate
    @Nullable
    private LocalDateTime updatedAt;
}
